### PR TITLE
Fix ADAMW, and track the loss

### PR DESCRIPTION
**Pull request by mcabbott**
This makes the same changes as https://github.com/FluxML/Flux.jl/pull/1612, xref #38.

It also alters the tests to keep track of the loss. Although it does not fix any of the convergence problems.

### CODE REVIEW

File: src/rules.jl [hunk 1]
```diff
@@ -405,7 +405,7 @@ weight decay regularization.
                          (no need to change default)
 """
 ADAMW(η = 1f-3, β = (9f-1, 9.99f-1), γ = 0, ϵ = eps(typeof(η))) =
-  OptimiserChain(ADAM{typeof(η)}(η, β, ϵ), WeightDecay(γ))
+  OptimiserChain(ADAM{typeof(η)}(1, β, ϵ), WeightDecay{typeof(η)}(γ), Descent(η))
```
Comment (ToucheSir) [inline src/rules.jl:408]:
```suggestion
  OptimiserChain(ADAM{typeof(η)}(1, β, ϵ), WeightDecay{typeof(η)}(γ), Descent(η))
```
Sorry, I just remembered a previous comment (maybe on Slack? Can't find it now) which noted that our implementation did not match the [paper](https://arxiv.org/abs/1711.05101v3):
image.

TL;DR is that (possibly because of an odd choice of variable names in the paper, they use `α` instead of `η` for the learning rate and `η` for something else), the "fixed" version is wrong and the original is correct. Apologies for the noise.

Comment (CarloLucibello) [reply src/rules.jl:408]: oh damn, Brian is right, https://github.com/FluxML/Flux.jl/pull/1612 should be reverted, I got confused by the nomenclature

Comment (CarloLucibello) [reply src/rules.jl:408]: I'll file a PR in Flux
